### PR TITLE
Add boost 1.74

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **boost/1.74**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/2524

Changelog: https://www.boost.org/users/history/version_1_74_0.html

STLInterfaces was added, but it's header-only: https://www.boost.org/doc/libs/1_74_0/more/getting_started/unix-variants.html#header-only-libraries

### CODE REVIEW

File: recipes/boost/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.1)
```
Comment (mathbunnyru) [inline recipes/boost/all/test_package/CMakeLists.txt:1]: To make hook error go away.

**Comment by mathbunnyru** [comment]
I had to create very similar patches to the existing ones, because line numbers have changed.

**Comment by AlexeyDmitriev** [comment]
closes #2524

**Comment by mathbunnyru** [comment]
I have no idea what's going on. I can't see what it means 'This commit cannot be built', because I don't see the logs

Could someone help me please? @SSE4 @madebr @Artalus

**Comment by Artalus** [comment]
Logs are above, in the message from conan-center-bot.
In first case there is some CI cuplrit apparently:
```
boost/1.73.0: ERROR: Unable to remove source folder C:/J/c3ipr/s/PR-2600/1/12_10\684d3b\1
Cannot change permissions for C:/J/c3ipr/s/PR-2600/1/12_10\684d3b\1\boost_1_73_0\boost\type_index! Exception info: (<class 'PermissionError'>, PermissionError(13, 'The process cannot access the file because it is being used by another process'), <traceback object at 0x0000011631E58E48>)
boost/1.73.0: WARN: **** Please delete it manually ****
ERROR: Unable to remove source folder
```
In second and third there seemingly something wrong with `b2` files:
```
C:\J\c3ipr\s\PR-2600\1\12_8\147788\1\source_subfolder\tools\build\src/kernel/bootstrap.jam:89: in boost-build

IMPORT error: rule "import" unknown in module "modules".
```

**Comment by mathbunnyru** [comment]
The logs above do correspond to the first try. After that I made a small change to actually build 1.74 and I can't see what's wrong with that.

**Comment by AlexeyDmitriev** [comment]
It would be really helpful if somebody from the devs could check what is failing here.
Because it appears to be impossible to check what is the error is for the last commit

Thanks

**Comment by SSE4** [comment]
@danimtb bot is behaving weird here...

**Review by danimtb — APPROVED** [review]

**Pull request merged**